tsfc/impero.py

```diff
@@ -90,13 +90,13 @@ class Return(Terminal):
     __front__ = ('variable', 'expression')
 
     def __init__(self, variable, expression):
-        assert set(variable.free_indices) == set(expression.free_indices)
+        assert set(variable.free_indices) >= set(expression.free_indices)
```

Comment:
OK, so now the lvalue of a return is allowed to have free indices that the rvalue does not.